include/DataFile.h

```diff
@@ -148,6 +122,13 @@ class LMMS_EXPORT DataFile : public QDomDocument
 	Type m_type;
 	unsigned int m_fileVersion;
 
+	template<unsigned int i, class Up>
+	void upgrade() {
+		if (i >= m_fileVersion) {
+			Up{*this}();
+		}
+	}
```

Comment:
The braces should be on a single line. I would also `inline` this function.

Reply:
It's already inlined implicitly bc it's in a class body, but your right about the brace.

Reply:
> It's already inlined implicitly bc it's in a class body.

You're right, I kind of just have a habit of making stuff `inline`, but you don't have to (and shouldn't since its redundant).